src/Microsoft.DotNet.VersionTools/Automation/LocalVersionsRepoUpdater.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
I looked at it earlier, briefly--part of the pain was that `NupkgNameInfo` is `internal`. I took a fresh look and decided to share functions as you suggested and just make `NupkgNameInfo` (now `NupkgInfo`) public.